host-bin/edit-chroot

```diff
@@ -335,15 +363,34 @@ for NAME in "$@"; do
                 dest="$dest.gz"
             fi
         fi
+        # If we're writing to a fat32 filesystem, split the file at 4GB chunks
+        if ! df -T "`getmountpoint "$dest"`" | awk '$2~"^v?fat"{exit 1}'; then
+            SPLIT="${SPLIT:-4GB}"
+        fi
+        if [ -n "$SPLIT" ]; then
+            tardest="`mktemp -u --tmpdir=/tmp 'crouton-backup.XXX'`"
+            mkfifo -m 600 "$tardest"
+            addtrap "rm -f '$tardest'"
+            split -b "$SPLIT" -d "$tardest" "$dest.part" &
+            splitpid="$!"
+        else
+            tardest="$dest"
+            splitpid=''
+        fi
         echo -n "  Backing up $CHROOT to $dest..." 1>&2
-        addtrap "echo 'Deleting partial archive.' 1>&2; rm -f '$dest'"
+        addtrap "echo 'Deleting partial archive.' 1>&2; \
+                 kill '$splitpid' 2>/dev/null; rm -f '$dest' '$dest.part'*"
         ret=0
         spinner 1 tar --checkpoint=100 --checkpoint-action=exec=echo \
             --one-file-system -V "crouton:backup.${date%-*}${date#*-}-$NAME" \
-            -caf "$dest" -C "$CHROOTS" "$NAME" || ret=$?
-        if [ "$ret" != 0 ]; then
+            -caf "$tardest" -C "$CHROOTS" "$NAME" || ret="$?"
```

Comment:
Similar remark as below, `-a` won't work as indented, and all split archives will be uncompressed.